app/core/effect.ts

```diff
@@ -20,17 +20,35 @@ export class OnItemGainedEffect extends Effect {}
 
 export class OnItemRemovedEffect extends Effect {}
 
-export class PeriodicEffect extends Effect {
+// applied after knocking out an enemy
+export class OnKillEffect extends Effect {
+  apply: (entity: PokemonEntity, target: PokemonEntity, board: Board) => void
+  constructor(
+    effect: (entity: PokemonEntity, target: PokemonEntity, board: Board) => void
+  ) {
+    super(effect)
+    this.apply = effect
+  }
+}
+
+export abstract class DynamicEffect {
```

Comment:
i dont understand the reasoning behind having another abstract class, Effect & DynamicEffect. If its just about the constructor signature to have the apply logic defined in a prototype method vs an own property passed in constructor, then just make the argument in the constructor optional